src/System.Private.CoreLib/src/System/Runtime/Loader/ComponentDependencyResolver.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Internal.IO;
+
+namespace System.Runtime.Loader
+{
+    public sealed class ComponentDependencyResolver
+    {
+        /// <summary>
+        /// The name of the neutral culture (same value as in Variables::Init in CoreCLR)
+        /// </summary>
+        private const string NeutralCultureName = "neutral";
+
+        /// <summary>
+        /// The extension of resource assembly (same as in BindSatelliteResourceByResourceRoots in CoreCLR)
+        /// </summary>
+        private const string ResourceAssemblyExtension = ".dll";
+
+        private readonly string _componentAssemblyPath;
+        private readonly Dictionary<string, string> _assemblyPaths;
+        private readonly string[] _nativeSearchPaths;
+        private readonly string[] _resourceSearchPaths;
+        private readonly string[] _componentDirectorySearchPaths;
+
+        public ComponentDependencyResolver(string componentAssemblyPath)
+        {
+            _componentAssemblyPath = componentAssemblyPath;
+
+            string assemblyPathsList = null;
+            string nativeSearchPathsList = null;
+            string resourceSearchPathsList = null;
+            int returnCode = 0;
+
+            try
+            {
+                returnCode = corehost_resolve_component_dependencies(
+                    componentAssemblyPath,
+                    (assembly_paths, native_search_paths, resource_search_paths) =>
+                    {
+                        assemblyPathsList = assembly_paths;
+                        nativeSearchPathsList = native_search_paths;
+                        resourceSearchPathsList = resource_search_paths;
+                    });
+            }
+            catch (EntryPointNotFoundException entryPointNotFoundException)
+            {
+                throw new InvalidOperationException(SR.ComponentDependencyResolver_FailedToLoadHostpolicy, entryPointNotFoundException);
+            }
+            catch (DllNotFoundException dllNotFoundException)
+            {
+                throw new InvalidOperationException(SR.ComponentDependencyResolver_FailedToLoadHostpolicy, dllNotFoundException);
```

Comment:
Seems like these two should throw a `NotSupportedException`

Reply:
I guess it might make sense... it's true that these would be the "fallback" cases if we decide to implement some fallback behavior.

Reply:
I guess I was thinking that the host was not supporting this....